homeassistant/components/watergate/sensor.py

```diff
@@ -0,0 +1,302 @@
+            if data.state and data.state.water_meter
+            else None,
+            UnitOfTime.MINUTES,
+            SensorDeviceClass.DURATION,
+        ),
+        SonicSensor(
+            coordinator,
+            IP_SENSOR_NAME,
+            IP_ENTITY_NAME,
+            lambda data: data.networking.ip if data.networking else None,
+            entity_category=EntityCategory.DIAGNOSTIC,
+        ),
+        SonicSensor(
+            coordinator,
+            GATEWAY_SENSOR_NAME,
+            GATEWAY_ENTITY_NAME,
+            lambda data: data.networking.gateway if data.networking else None,
+            entity_category=EntityCategory.DIAGNOSTIC,
+        ),
+        SonicSensor(
+            coordinator,
+            SUBNET_SENSOR_NAME,
+            SUBNET_ENTITY_NAME,
+            lambda data: data.networking.subnet if data.networking else None,
+            entity_category=EntityCategory.DIAGNOSTIC,
+        ),
+        SonicSensor(
+            coordinator,
+            SSID_SENSOR_NAME,
+            SSID_ENTITY_NAME,
+            lambda data: data.networking.ssid if data.networking else None,
+            entity_category=EntityCategory.DIAGNOSTIC,
+        ),
+        SonicSensor(
+            coordinator,
+            RSSI_SENSOR_NAME,
+            RSSI_ENTITY_NAME,
+            lambda data: data.networking.rssi if data.networking else None,
+            SIGNAL_STRENGTH_DECIBELS_MILLIWATT,
+            SensorDeviceClass.SIGNAL_STRENGTH,
+            EntityCategory.DIAGNOSTIC,
+        ),
+        SonicSensor(
+            coordinator,
+            WIFI_UPTIME_SENSOR_NAME,
+            WIFI_UPTIME_ENTITY_NAME,
+            lambda data: data.networking.wifi_uptime if data.networking else None,
+            UnitOfTime.MILLISECONDS,
+            SensorDeviceClass.DURATION,
+            EntityCategory.DIAGNOSTIC,
+        ),
+        SonicSensor(
+            coordinator,
+            MQTT_UPTIME_SENSOR_NAME,
+            MQTT_UPTIME_ENTITY_NAME,
+            lambda data: data.networking.mqtt_uptime if data.networking else None,
+            UnitOfTime.MILLISECONDS,
+            SensorDeviceClass.DURATION,
+            EntityCategory.DIAGNOSTIC,
+        ),
+        SonicSensor(
+            coordinator,
+            TEMPERATURE_SENSOR_NAME,
+            TEMPERATURE_ENTITY_NAME,
+            lambda data: data.telemetry.water_temperature if data.telemetry else None,
+            UnitOfTemperature.CELSIUS,
+            SensorDeviceClass.TEMPERATURE,
+        ),
+        SonicSensor(
+            coordinator,
+            PRESSURE_SENSOR_NAME,
+            PRESSURE_ENTITY_NAME,
+            lambda data: data.telemetry.pressure if data.telemetry else None,
+            UnitOfPressure.MBAR,
+            SensorDeviceClass.PRESSURE,
+        ),
+        SonicSensor(
+            coordinator,
+            FLOW_SENSOR_NAME,
+            FLOW_ENTITY_NAME,
+            lambda data: (data.telemetry.flow / 1000)
```

Comment:
I think if you use `unit_of_measurement` and `native_unit_of_measurement` HA will do the transformation for you.